### PR TITLE
Add open api spec for metadata_service

### PR DESCRIPTION
**Pull request by dmateusp**
### Summary of Changes

Add OpenAPI (swagger) docs to metadata_service, https://github.com/lyft/amundsen/issues/166

### Tests

Brought the swagger_tests from the search_service with minor modifications (I changed the .replace() calls to cover cases with `<path:something>` inside the URL)

### Documentation

I added a reference to the API docs in the README.md

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

**Comment by dmateusp** [comment]
Looking into adding tests and documentation from search_service now

File: metadata_service/api/user.py [hunk 1]
```diff
@@ -50,15 +52,15 @@ def get(self, user_id: str) -> Iterable[Union[Mapping, int, None]]:
             return {'message': 'User id {} does not exist'.format(user_id)}, HTTPStatus.NOT_FOUND
 
 
-class UserFollowAPI(Resource):
+class UserFollowsAPI(Resource):
```
Comment (feng-tao) [inline metadata_service/api/user.py]: why do we need to add new class for get instead of putting get/put etc under a same resource?

Comment (dmateusp) [reply metadata_service/api/user.py]: I think it is described in the PR that was linked in the issue you raised: https://github.com/lyft/amundsensearchlibrary/pull/60 (I was seeing the same issue in metadata service where some URLs were invalid for some methods)

> [...] However, the Swagger documentation is looking at all the endpoints and methods setup in Flask, so this bug was causing the Swagger documentation to give users bad information. 😞

> Why is this happening?

> These lines of code in the service init file are the problem. This code is adding the resource to Flask and allowing all methods to be accessed via both URLs. However, both URLs are not supported by our methods.

In summary, each class only has "one type of URL"

Comment (feng-tao) [reply metadata_service/api/user.py]: ah, forgot about that:) thanks.

**Pull request merged**